### PR TITLE
Use standalone policies to self-manage credentials

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR:
 * Creates two new standalone IAM policies:
   * One that allows users to administer their own user accounts, while requiring multi-factor authentication (MFA)
   * One that allows users to administer their own user accounts, without requiring MFA
 * Replaces the inline policy attached to godlike users with the standalone (non-MFA) policy above
 

## 💭 Motivation and context ##

The main goal here is to create these credential management policies in a central location and make them available for use elsewhere in other repositories.  See https://github.com/cisagov/cool-users-non-admin/pull/18 for the first external usage of these policies.

## 🧪 Testing ##

I applied these changes in Production and verified via the AWS web console that the correct policies were created and attached to each godlike user.  I also confirmed that my user was still able to assume a role programmatically, which meant that the new policy was working as intended.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]
"Sound goods to my's ears!" -Skwisgaar Skwigelf

File: users/selfmanagedcredswithoutmfa_policy.tf [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  # Allow users to administer their own signing certificates
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "iam:DeleteSigningCertificate",
+      "iam:ListSigningCertificates",
+      "iam:UpdateSigningCertificate",
+      "iam:UploadSigningCertificate",
+    ]
```
Comment (felddy) [inline users/selfmanagedcredswithoutmfa_policy.tf:72]: Can you remind me what these certs are used for?

File: users/selfmanagedcredswithoutmfa_policy.tf [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      "iam:UpdateSSHPublicKey",
+      "iam:UploadSSHPublicKey",
+    ]
+
+    resources = [
+      "arn:aws:iam::*:user/&{aws:username}",
+    ]
+  }
+
+  # Allow users to administer their own git credentials
```
Comment (felddy) [inline users/selfmanagedcredswithoutmfa_policy.tf:96]: Is this truly just for `git`.

**Review by felddy — APPROVED** [review]
Looks good.  I had a few questions to help me understand a few thangs.

💪

File: users/selfmanagedcredswithoutmfa_policy.tf (hunk 2 above)
Comment (jsf9k) [reply users/selfmanagedcredswithoutmfa_policy.tf:96]: The only use I know of is AWS CodeCommit.

File: users/selfmanagedcredswithoutmfa_policy.tf (hunk 1 above)
Comment (jsf9k) [reply users/selfmanagedcredswithoutmfa_policy.tf:72]: I believe you can use a signing certificate as an alternative to the AWS programmatic credentials.

Comment (dav3r) [reply users/selfmanagedcredswithoutmfa_policy.tf:72]: I don't believe we currently use any signing certificates, but I'm not 100% sure.  As a reminder, the contents of this policy are borrowed from [this AWS example](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws_my-sec-creds-self-manage.html) - this is mentioned in the comment at the top of our policy file:
https://github.com/cisagov/cool-accounts/blob/7073cdc641ab91adde6bff70aed1912eeb800475/users/selfmanagedcredswithoutmfa_policy.tf#L2-L4

File: users/selfmanagedcredswithoutmfa_policy.tf (hunk 2 above)
Comment (dav3r) [reply users/selfmanagedcredswithoutmfa_policy.tf:96]: From [this page](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws_my-sec-creds-self-manage.html):

> [this] statement allows the user to create, update, and delete their own Git credentials for CodeCommit.

We don't yet use this feature anywhere either, as far as I am aware.

More info here:
* https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateServiceSpecificCredential.html
* https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_ssh-keys.html

File: users/selfmanagedcredswithoutmfa_policy.tf (hunk 1 above)
Comment (jsf9k) [reply users/selfmanagedcredswithoutmfa_policy.tf:72]: We don't use any signing certificates.

**Pull request merged**